app/routes.py

```diff
@@ -1,2 +1,231 @@
-from flask import Blueprint
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, request, abort, make_response
+from datetime import datetime, timezone
+import os
+from dotenv import load_dotenv
+import requests
 
+tasks_bp = Blueprint("task_list", __name__, url_prefix="/tasks")
+
+def valid_int(number, parameter_type):
+    try:
+        int(number)
+    except:
+        abort(make_response({"error": f"{parameter_type} must be an int"})), 400
+
+def get_task_from_id(task_id):
+    valid_int(task_id, "task_id")
+    return Task.query.get_or_404(task_id, description="{task not found}")
+
+@tasks_bp.route("", methods=["POST"])
+def post_new_task():
+    
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body\
+         or "completed_at" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+
+    new_task = Task(title=request_body["title"],
+    description=request_body["description"],
+    completed_at=request_body["completed_at"])
+   
+    db.session.add(new_task)
+    db.session.commit()
+
+    response_body = {
+        "task": (new_task.to_dict())
+    }
+    return jsonify(response_body), 201
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def getting_one_task(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return jsonify(None), 404
+    
+    response_body = {
+        "task": (task.to_dict())
+    }
+    return jsonify(response_body), 200
+
+@tasks_bp.route("", methods=["GET"])
+def getting_tasks():
+    query = Task.query # Base query
+
+    # Query params, adding to query where indicated
+    sort = request.args.get("sort")
+    if sort == "asc":
+        query = query.order_by(Task.title)
+    elif sort == "desc":
+        query = query.order_by(Task.title.desc())
+    
+    query = query.all() # Final query
```

Comment:
Consider naming query `tasks` to enhance readability.
```suggestion
    tasks = query.all() # Final query
```
```suggestion
    query = query.all() # Final query
```